frontend/src/components/MapMarker.tsx

```diff
@@ -5,9 +5,26 @@ import { type City } from '../api.generated';
 export default function MapMarker(props: {
   city: City;
 }): React.ReactElement {
+  // This function => downloads the discription in the text file format
+  const downloadDescriptionFunc = () : void =>{
+    const descString = props.city.shortDescription;
+    const filename = 'description.txt';
+    const element = document.createElement('a');
+    const file = new Blob([descString], {type: 'text/plain'});
+    element.href = URL.createObjectURL(file);
+    element.download = filename;
+    document.body.appendChild(element);
+    element.click();
+    document.body.removeChild(element);
+  }
   return (
     <Marker position={{ lat: props.city.lat, lng: props.city.lng }}>
-      <Popup>{props.city.shortDescription}</Popup>
+      <Popup>
+        <p>
+          {props.city.shortDescription}
+        </p>
+        <button onClick={downloadDescriptionFunc} >Download description</button>
```

Comment:
We use Material UI as a component library for this project. Why is it necessary to use a plain html button here?

This way it will also not adapt to any theming we introduce later via the MUI theming api

Reply:
used the MUI components.